### DISCUSSION TITLE
In a store, how do you access Nuxt 3 context?

### DISCUSSION THREAD

**Discussion by nathanchase** [nuxt/nuxt#17215] [Questions] [answered]
In Nuxt 3, I can get to `$auth` ([auth module](https://github.com/Teranode/nuxt-module-alternatives/tree/master/%40nuxtjs-alt/auth)) by way of `useNuxtApp()` in components and composables, but how would you access it in a [Pinia](https://github.com/vuejs/pinia) store?

In components, you can do:
```js
const nuxtApp = useNuxtApp();
const $auth = nuxtApp.$auth; // instance of the auth module
const authUser = $auth.user; // user property of the auth module
```
but in a `stores/store.ts`, `useNuxtApp()` is not available (nor is `this.$auth`), so I'm not quite sure how to get to the context?
```
[nuxt] [request error] nuxt instance unavailable
  at Module.useNuxtApp (./.nuxt/dist/server/server.mjs:520:13)
  at $id_C1ltjrBOrk (./.nuxt/dist/server/server.mjs:10547:39)
  at async __instantiateModule__ (./.nuxt/dist/server/server.mjs:32909:3)
```
References:
https://github.com/Teranode/nuxt-module-alternatives/discussions/54
https://github.com/vuejs/pinia/pull/1433

**Comment by bros4president** [comment, marked as answer]
Starting from rc.5 and @pinia/nuxt 0.3.0, `useNuxtApp` is available in stores.

You have to call it inside `definePiniaStore` though, for example in an action. Otherwise you’ll get `nuxt instance is unavailable`, because top level plugin code will be run too early in the framework lifecycle.

Before rc.5 and previous @pinia/nuxt, `this.$nuxt` was available inside actions. It referenced the legacy context rc.5 dropped :)

**Reply by nathanchase** [reply]
Ok, that was my issue. I was trying to do `const nuxtApp = useNuxtApp();` in the store file at the top below the imports - not inside of each individual getter, or action. Thank you!
